packages/flutter_tools/lib/src/commands/install.dart

```diff
@@ -52,19 +49,24 @@ class InstallCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    final ApplicationPackage package = await applicationPackages.getPackageForPlatform(
-      await device.targetPlatform,
+    final Device targetDevice = device!;
+    final ApplicationPackage? package = await applicationPackages?.getPackageForPlatform(
+      await targetDevice.targetPlatform,
+      buildInfo: BuildInfo.debug, // If app is not pre-built, default to building debug.
```

Comment:
Is this a change in behavior?

Reply:
Good catch, I should have commented on this.

It's only used when there isn't a pre-built iOS app.  
https://github.com/flutter/flutter/blob/9e88fe328efcf42d3037a965b39269f0805cd4ce/packages/flutter_tools/lib/src/flutter_application_package.dart#L83-L84
Previously, this was passing null all the way through the iOS project properties fetches, and that had been converted to null safety and was expecting a non-null `buildInfo`.  
https://github.com/flutter/flutter/blob/43732e45977a04e662f0d4b2612b5f95678eaa74/packages/flutter_tools/lib/src/xcode_project.dart#L219

Fortunately, it was still handling null `buildInfo`s and was falling back to default values, which work for most iOS Flutter apps:
https://github.com/flutter/flutter/blob/43732e45977a04e662f0d4b2612b5f95678eaa74/packages/flutter_tools/lib/src/ios/xcodeproj.dart#L438

The next thing that happens is complaining that the bundle isn't pre-built anyway:
```
Could not find application bundle at build/ios/iphoneos/Runner.app; have you run "flutter build ios"?
```
What I have here passing in `BuildInfo.debug` is more correct since it's possible `Runner.app` isn't the correct bundle name if you've opened your Xcode project and futzed around with certain build settings, particularly if you are using flavors.

However, since null `buildInfo`s are handled all the way down, and I don't want to change the behavior in this PR, I'll change it back and we can actually fix this if real users ever complain (I don't see an open issue for it).